### PR TITLE
wicket-6.x: More OpenLayer3 Changes

### PR DESCRIPTION
**Pull request by cmiles74**
These are the same changes included in my pull request for the master branch, these are applied to the wicket-6.x branch.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,35 +2,4 @@ language: java
...
-  - '[ "${TRAVIS_PULL_REQUEST}" = "false" ] && DEPLOY_OR_TEST=deploy'
-  - "echo DEPLOY_OR_TEST=$DEPLOY_OR_TEST"
-  - "jdk_switcher use openjdk6"
-  - "mvn -s sonatype-settings.xml clean $DEPLOY_OR_TEST -N $MVN_OPTS"
-  - "cd jdk-1.6-parent"
-  - "mvn -s ../sonatype-settings.xml clean $DEPLOY_OR_TEST $MVN_OPTS"
-  - "cd ../jdk-1.7-parent"
-  - "jdk_switcher use oraclejdk7"
-  - "mvn -s ../sonatype-settings.xml clean $DEPLOY_OR_TEST $MVN_OPTS"
+    - twitch@nervestaple.com
```
Comment (martin-g) [inline .travis.yml]: Please add one more commit to revert the changes here.

Comment (martin-g) [inline .travis.yml]: Another way is to use `git rebase --interactive THE_COMMIT_BEFORE_YOUR_CHANGES` and then delete the line of the commit that touched .travis.yml. It will work only if this commit doesn't touch other files. Otherwise you'd lose the meaningful changes too.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+[![Build Status](https://travis-ci.org/cmiles74/core.svg?branch=wicket-6.x)](https://travis-ci.org/cmiles74/core)
```
Comment (martin-g) [inline README.md]: Remove please

File: jdk-1.6-parent/openlayers3-parent/openlayers3-bootstrap/src/main/java/org/wicketstuff/openlayers3/api/overlay/Popover.java [hunk 3]
```diff
@@ -248,6 +253,16 @@ private String renderPopoverJs() {
 
         builder.append("});");
 
+        if(titleModel != null) {
+            builder.append("$(element).attr('data-original-title', '" + escapeQuoteJs(titleModel.getObject())  + "');" );
```
Comment (martin-g) [inline jdk-1.6-parent/openlayers3-parent/openlayers3-bootstrap/src/main/java/org/wicketstuff/openlayers3/api/overlay/Popover.java]: Use `.append()` instead of concatenation.

Comment (martin-g) [inline jdk-1.6-parent/openlayers3-parent/openlayers3-bootstrap/src/main/java/org/wicketstuff/openlayers3/api/overlay/Popover.java]: Use `.data('original-title')` instead of `.attr('data-original-title')`.

File: jdk-1.6-parent/openlayers3-parent/openlayers3-bootstrap/src/main/java/org/wicketstuff/openlayers3/api/overlay/PopoverOverlay.java [hunk 4]
```diff
@@ -16,4 +16,9 @@
     public PopoverOverlay(MarkerPopover markerPopover) {
         super(markerPopover, markerPopover.getPositionModel().getObject(), DEFAULT_POSITIONING, DEFAULT_STOP_EVENT);
     }
+
+    @Override
+    public String getJsId() {
+        return "popover_" + "overlay_" + element.getMarkupId();
```
Comment (martin-g) [inline jdk-1.6-parent/openlayers3-parent/openlayers3-bootstrap/src/main/java/org/wicketstuff/openlayers3/api/overlay/PopoverOverlay.java]: no need of the first concatenation.

File: jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/BingPage.html [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                <p>Below is a map of the world. This map uses a Bing Maps source to provide the image tiles. This map is
+                    centered on the location of my office.</p>
+
+                <div wicket:id="map" style="height: 600px; margin-top: 1em; border: 1px solid #a9a9a9">[MAP]</div>
+
+                <hr>
+
+                <ul>
+                    <li>
+                        <a href="https://github.com/wicketstuff/core/blob/wicket-6.x/jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/BingPage.java">
```
Comment (martin-g) [inline jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/BingPage.html]: `target="_blank"`

File: jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/BingPage.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+import org.wicketstuff.openlayers3.api.coordinate.LongLat;
+import org.wicketstuff.openlayers3.api.layer.Layer;
+import org.wicketstuff.openlayers3.api.layer.Tile;
+import org.wicketstuff.openlayers3.api.source.BingMaps;
+import org.wicketstuff.openlayers3.examples.base.BasePage;
+
+import java.util.Arrays;
+
+/**
+ * Provides a page with a map that uses OpenStreetMap tiles.
```
Comment (martin-g) [inline jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/BingPage.java]: Please update the javadoc with something like `... and uses M$ Bing...`

File: jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/ModifyPage.html [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+                <p>Miles' Office is at <span wicket:id="featureLocationLabel">[LOCATION]</span>.</p>
+
+                <div wicket:id="map" style="height: 600px; margin-top: 1em; border: 1px solid #a9a9a9">[MAP]</div>
+
+                <hr>
+
+                <ul>
+                    <li>
+                        <a href="https://github.com/wicketstuff/core/blob/wicket-6.x/jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/ModifyPage.java">
```
Comment (martin-g) [inline jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/ModifyPage.html]: target=blank

File: jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/ModifyPage.java [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+            protected String load() {
+
+                StringBuilder builder = new StringBuilder();
+
+                if(featureLocationModel.getObject() != null) {
+                    builder.append(featureLocationModel.getObject().getX());
+                    builder.append(", ");
+                    builder.append(featureLocationModel.getObject().getY());
+                } else {
+                    builder.append("Unkown...");
```
Comment (martin-g) [inline jdk-1.6-parent/openlayers3-parent/openlayers3-examples/src/main/java/org/wicketstuff/openlayers3/examples/ModifyPage.java]: typo

**Comment by cmiles74** [comment]
Thank you for looking this over, I have made these changes to both pull requests. :-)

File: .travis.yml [hunk 9]
```diff
@@ -22,15 +24,8 @@ cache:
...
+  - "DEPLOY_OR_TEST=install"
   - '[ "${TRAVIS_PULL_REQUEST}" = "false" ] && DEPLOY_OR_TEST=deploy'
   - "echo DEPLOY_OR_TEST=$DEPLOY_OR_TEST"
-  - "jdk_switcher use openjdk6"
-  - "mvn -s sonatype-settings.xml clean $DEPLOY_OR_TEST -N $MVN_OPTS"
-  - "cd jdk-1.6-parent"
-  - "mvn -s ../sonatype-settings.xml clean $DEPLOY_OR_TEST $MVN_OPTS"
-  - "cd ../jdk-1.7-parent"
-  - "jdk_switcher use oraclejdk7"
-  - "mvn -s ../sonatype-settings.xml clean $DEPLOY_OR_TEST $MVN_OPTS"
```
Comment (martin-g) [inline .travis.yml]: The changes in this file are still wrong.
Most of the modules should be built with JDK 6. Just a few with JDK 7.
Please revert to the initial state before your changes.

**Comment by martin-g** [comment]
I did it.
Thanks for the contributions and sorry for the delay!
